manifests/params.pp

```diff
@@ -34,7 +34,24 @@
   $postrun_command     = undef
   $dns_alt_names       = []
   $use_srv_records     = false
-  $srv_domain          = $::domain
+
+  # On versions < 3.7.5 there was no defined function as such strict variables cannot be used
+  # On verisons < 4.0.0 and >= 3.5.0 the facts hash was optional disabled by default, on version < 3.5.0 the facts hash did not exist
+  # In some edge cases 'domain' fact may be not defined
+  if versioncmp($::puppetversion, '3.7.5') < 0 {
+    $srv_domain = $::domain
+  } elsif versioncmp($::puppetversion, '4.0.0') >= 0 {
```

Comment:
> As such while yes defined helps insulate when strict vars is enabled allowing for a successful compile, but because of the absence of the variable entirely it causes a flooding of the puppetserver logs regardless of strict var being enabled or disabled...

Which line causes this? The `defined` check itself? I couldn't reproduce this locally.